pyega3/pyega3.py

```diff
@@ -38,606 +33,707 @@
+                        output_file = None
+                    else:
+                        output_file = self.generate_output_filename(output_dir,
+                                                                    res['fileId'],
+                                                                    res['displayFileName'],
+                                                                    genomic_range_args)
+
+                    self.download_file_retry(
+                        credentials, res['fileId'], res['displayFileName'], res['fileName'], res['fileSize'],
+                        res['unencryptedChecksum'],
+                        num_connections, key, output_file, genomic_range_args, max_retries, retry_wait)
+            except Exception as ex:
+                logging.exception(ex)
+
+    @staticmethod
+    def print_debug_info(url, reply_json, *args):
+        logging.debug("Request URL : {}".format(url))
+        if reply_json is not None:
+            logging.debug("Response    :\n %.1200s" % json.dumps(reply_json, indent=4))
+
+        for arg in args:
+            logging.debug(arg)
+
+    @staticmethod
+    def create_parser():
+        parser = argparse.ArgumentParser(description="Download from EMBL EBI's EGA (European Genome-phenome Archive)")
+        parser.add_argument("-d", "--debug", action="store_true", help="Extra debugging messages")
+        parser.add_argument("-cf", "--config-file", dest='config_file',
+                            help='JSON file containing credentials/config e.g.{"username":"user1","password":"toor"}')
+        parser.add_argument("-sf", "--server-file", dest='server_file',
+                            help='JSON file containing server config e.g.{"url_auth":"aai url","url_api":"api url", '
+                                 '"url_api_ticket":"htsget url", "client_secret":"client secret"}')
+        parser.add_argument("-c", "--connections", type=int, default=1,
+                            help="Download using specified number of connections")
+        parser.add_argument("-t", "--test", action="store_true", help="Test user activated")
+
+        subparsers = parser.add_subparsers(dest="subcommand", help="subcommands")
+
+        subparsers.add_parser("datasets", help="List authorized datasets")
+
+        parser_dsinfo = subparsers.add_parser("files", help="List files in a specified dataset")
+        parser_dsinfo.add_argument("identifier", help="Dataset ID (e.g. EGAD00000000001)")
+
+        parser_fetch = subparsers.add_parser("fetch", help="Fetch a dataset or file")
+        parser_fetch.add_argument("identifier", help="Id for dataset (e.g. EGAD00000000001) "
+                                                     "or file (e.g. EGAF12345678901)")
+
+        parser_clean = subparsers.add_parser("clean", help="Deletes those temporary files "
+                                                           "which were created during the download process.")
```

Comment:
Suggest help="Delete temporary files created during download."

This suggestion is to match the brevity and verb tense of the other subcommands.

Reply:
What should be the name of the sub-command? Currently it's called "clean".

Currently, the user can execute "pyega3 clean --dir=/where/to/clean" and "pyega3 fetch --delete-temp-files EGAF0123". So we have "clean" and "--delete-temp-files". Should I synchronise them?

What should be the name of the "--dir" flag of the "clean" sub-command? Is "dir" OK or is there a better name for it?

Reply:
`--directory DIR, -dir DIR` could also work for the clean subcommand. I know it's customary to only have a single character after `-`, but there is already the `-d` argument for debugging. I think it will be less confusing and more clear to have `--directory DIR, -dir DIR` here.

Reply:
I think `clean` subcommand name is fine. I can see a case for `--clean-temp-files` as an argument to the `fetch` subcommand. In both cases, the help message for `clean` and `--clean-temp-files` indicate that files are deleted, so it should be clear to users what "clean" means.

Reply:
1. I renamed the arguments for the clean subcommand to `--directory DIR, -dir DIR`.

1. I left the `clean` subcommand as `clean`.
   I also left the `--delete-temp-files` argument as `--delete-temp-files`, and didn't change it to `--clean-temp-files`.

1. I changed the help message of the `--delete-temp-files` argument (of the `fetch` subcommand) from:
   ```
    --delete-temp-files   Delete those temporary, partial files which were
                          left on the disk after a failed transfer.
    ```
    to:
    ```
    --delete-temp-files   Do not keep those temporary, partial files which were
                          left on the disk after a failed transfer.
    ```
    @malloryfreeberg, could you please review that sentence? Thanks!

Reply:
LGTM!